### PR TITLE
deps: Update cert manager to 1.12.6

### PR DESCRIPTION
**Pull request by 3u13r**
### Proposed change(s)

- Bump cert manager to 1.12.6 (see `generateCertManager.sh)

### Additional info

- Any additional information or context

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

**Comment by daniel-weisse** [comment]
Why are we upgrading to v1.12 instead of the v1.13?
Are there any compatibility issues with that version?

**Comment by 3u13r** [comment]
> Why are we upgrading to v1.12 instead of v1.13?
Are there any compatibility issues with that version?

From https://cert-manager.io/docs/releases/upgrading/upgrading-1.12-1.13:
```
IMPORTANT NOTE: If upgrading from a version below v1.12, upgrade to the latest v1.12 release before upgrading to v1.13. Otherwise, some certificates may be unexpectedly re-issued (see https://github.com/cert-manager/cert-manager/issues/6494#issuecomment-1816112309)
```

File: internal/constellation/helm/helm_test.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func TestHelmApply(t *testing.T) {
 				awsLbVersion = *tc.clusterAWSLBVersion
 			}
 
-			certManagerVersion := "v1.10.0" // current version
+			certManagerVersion := "v1.13.2" // current version
```
Comment (daniel-weisse) [inline internal/constellation/helm/helm_test.go:197]:
```suggestion
			certManagerVersion := "v1.12.6" // current version
```
Not sure if this is relevant for the test

**Review by daniel-weisse — APPROVED** [review]
Can you run the upgrade test and make sure things still work?
lgtm otherwise

**Comment by 3u13r** [comment]
Azure, e2e upgrade, 3:2 : https://github.com/edgelesssys/constellation/actions/runs/7206377406

**Review by derpsteb — APPROVED** [review]

**Pull request merged**